dask/array/core.py

```diff
@@ -232,46 +233,88 @@ def slices_from_chunks(chunks):
     return list(product(*slices))
 
 
-def getem(
-    arr,
+def graph_from_arraylike(
+    arr,  # Any array-like which supports slicing
     chunks,
+    shape,
+    name,
     getitem=getter,
-    shape=None,
-    out_name=None,
     lock=False,
     asarray=True,
     dtype=None,
+    inline_array=True,
 ):
-    """Dask getting various chunks from an array-like
-
-    >>> getem('X', chunks=(2, 3), shape=(4, 6))  # doctest: +SKIP
-    {('X', 0, 0): (getter, 'X', (slice(0, 2), slice(0, 3))),
-     ('X', 1, 0): (getter, 'X', (slice(2, 4), slice(0, 3))),
-     ('X', 1, 1): (getter, 'X', (slice(2, 4), slice(3, 6))),
-     ('X', 0, 1): (getter, 'X', (slice(0, 2), slice(3, 6)))}
-
-    >>> getem('X', chunks=((2, 2), (3, 3)))  # doctest: +SKIP
-    {('X', 0, 0): (getter, 'X', (slice(0, 2), slice(0, 3))),
-     ('X', 1, 0): (getter, 'X', (slice(2, 4), slice(0, 3))),
-     ('X', 1, 1): (getter, 'X', (slice(2, 4), slice(3, 6))),
-     ('X', 0, 1): (getter, 'X', (slice(0, 2), slice(3, 6)))}
     """
-    out_name = out_name or arr
+    Dask getting various chunks from an array-like, which should support slicing.
+
+    If ``inline_array`` is True, this generates a HighLevelGraph with blockwise logic,
+    where the array-like is embedded in every task.
+    If ``inline_array`` is False, this generates a materialized graph where the
+    array-like is in a separate value in the graph, and other slicing tasks refer to it.
```

Comment:
```suggestion
    If ``inline_array`` is True, this makes a Blockwise layer of slicing tasks where
    the array-like is embedded in every task.
    If ``inline_array`` is False, this inserts the array-like as a standalone value in the
    graph, then generates a Blockwise layer of slicing tasks that refer to it.
```
Want to make it clear that the primary graph is blockwise in both cases